core/src/repair/repair_service.rs

```diff
@@ -677,6 +685,31 @@ impl RepairService {
         }
     }
 
+    pub(crate) fn generate_repairs_for_wen_restart(
+        blockstore: &Blockstore,
+        max_repairs: usize,
+        slots: Arc<RwLock<Vec<Slot>>>,
+    ) -> Vec<ShredRepairType> {
+        let mut result: Vec<ShredRepairType> = Vec::new();
+        for slot in slots.read().unwrap().iter() {
+            if let Some(slot_meta) = blockstore.meta(*slot).unwrap() {
+                let new_repairs = Self::generate_repairs_for_slot(
+                    blockstore,
+                    *slot,
+                    &slot_meta,
+                    max_repairs - result.len(),
+                );
+                result.extend(new_repairs);
+            } else {
+                result.push(ShredRepairType::Shred(*slot, 0));
```

Comment:
I'm guessing you want `HighestShred` here.

Reply:
Good catch, changed.